### PR TITLE
[1/5] Rename integration table to resource

### PR DESCRIPTION
**Pull request by kenxu95**
## Describe your changes and why you are making these changes
- Changes the name of our `integration` table to be `resource`. 
- Adds a migration for this name change. 
- Updates all references to the integration table.
- Also renames all "integration" found in `lib/models` to resource, EXCEPT for any json/yaml tags like 
```
	DestResourceID uuid.UUID             `db:"dest_integration_id" json:"dest_integration_id"`
```
That means we do a follow-up migration that updates all database field references to "integration", as well as any client data structures.

## Related issue number (if any)
ENG-2780

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/cmd/server/handler/create_table.go [hunk 1]
```diff
@@ -140,7 +140,7 @@ func CreateTable(
 	ctx context.Context,
 	args *CreateTableArgs,
 	contentPath string,
-	integrationObject *models.Integration,
+	integrationObject *models.Resource,
```
Comment (saurav-c) [inline src/golang/cmd/server/handler/create_table.go:143]: Do you want to change variable names as a follow on?

**Review by saurav-c — APPROVED** [review]
Is there a linear task for the JSON tag changes?

**Comment by saurav-c** [comment]
Also note to take a look at the DB integration tests

**Review by saurav-c — CHANGES_REQUESTED** [review]
@kenxu95 You'll need to replace the table name being deleted [here](https://github.com/aqueducthq/aqueduct/blob/main/src/golang/lib/repos/tests/suite.go#L111)

File: src/golang/cmd/server/handler/create_table.go (hunk 1 above)
Comment (saurav-c) [reply src/golang/cmd/server/handler/create_table.go:143]: Nvm saw follow on PR

**Comment by kenxu95** [comment]
> Is there a linear task for the JSON tag changes?

No task yet but that will be the 4th out of 5th step.

**Review by likawind — APPROVED** [review]

**Review by saurav-c — APPROVED** [review]

**Pull request merged**